### DISCUSSION TITLE
Labels as model

### DISCUSSION THREAD

**Discussion by cybercop23** [xLightsSequencer/xLights#4647] [General]
There was a question/request to add a new model, Label, that doesn't use any channels. Most (including myself) use the Image model to mark controller, receivers, power injection... so we can use that same model, add a String Property, Label and thus that will not take any channels and will even be removed from the models available in the visualizer.
I recall some similar conversations of maybe adding support for text boxes, may be nice, but harder to implement.

What do you think of a) supporting such feature & b) what would be the easiest or best way to implement this?

**Comment by AzGilrock** [comment]
I've always had the idea of adding a 3rd tab to be used for drawing 
things like controller images, wiring connections, etc.  So we have a 
models tab and an objects tab and there could be a 3rd tab.  The reason 
for a new tab is the same reason I did the objects tab and that was so 
that when you are selecting models things in other tabs can't be 
selected and don't get in the way.  You could provide an easy way to 
hide everything in the tab if needed.  I've thought it would be cool to 
be able to drag and drop all your controllers as objects into the scene 
and then use maybe PolyLines to draw wiring diagrams between them.

Regarding labels I thought the main reason we didn't have them yet was 
because there wasn't an easy way to draw text in OpenGL.  Maybe there 
are new options now but I thought when I looked it up years ago you had 
to create your own font set with all the line drawing commands to create 
each letter.

Gil

**Comment by cybercop23** [comment]
So a new tab is intertesting, but for the wiring diagram, you'll still need the full layout, no? So is it then worth adding one.
Right now, most of us have Daryl's icons and Rick creates images of text, but it will be cool if you could use the paint example you based the serialization on and allow users to just write.

**Comment by AzGilrock** [comment]
The new tab would already have the full layout its just another 
layer/overlay.  Like the Objects tab displays the house but when you are 
working with models you don't have to worry about the cursor trying to 
select the house object.

**Comment by keithsw1111** [comment]
I have a branch with text labels implemented but i could only ever get it working in 2D ... 3D was beyond my opengl skills

**Reply by cybercop23** [reply]
Is there a way to preview that? Is it just using text labels, like you would do in a powerpoint presentation?

**Comment by keithsw1111** [comment]
I pushed layout label branch. Feel free to take it forward. I had to spend some time rewriting a lot of it as it was all built pre-metal so a lot of the rendering has changed since then. It works as an object in 3d. I think i tried to get it working in 2d but had issues. Mainly because i didnt know how to project the text. I am sure it is possible just was beyond me then and likely is now as well.

![image](https://github.com/user-attachments/assets/78b3630f-8c47-4ac4-9df7-6a9e1ffd9263)
